sw/device/lib/testing/dif/mock_dif_uart_test.cc

```diff
@@ -0,0 +1,228 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/mock_mmio.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+
+extern "C" {
+#include "sw/device/lib/dif/dif_uart.h"
+#include "uart_regs.h"  // Generated.
+}  // extern "C"
+
+namespace dif_uart_test {
```

Comment:
I'd also take care to include an anonymous namespace here, as well, to prevent name leakage.